src/moonlink/src/storage/mooncake_table/snapshot_read_output.rs

```diff
@@ -71,50 +85,118 @@ impl ReadOutput {
             .unwrap();
     }
 
+    /// File ordering should be preserved because the index determines the position
+    /// in the final [`resolved_data_files`] array, which delete or other operations rely on.
+    async fn resolve_remote_files(
+        &mut self,
+        object_storage_cache: Arc<dyn CacheTrait>,
+        filesystem_accessor: Arc<dyn BaseFileSystemAccess>,
+        remote_files_entries: Vec<RemoteFileEntry>,
+        resolved_data_files: &mut [String],
+        cache_handles: &mut Vec<NonEvictableHandle>,
+    ) -> Result<()> {
+        let mut results = Vec::with_capacity(remote_files_entries.len());
+        let mut stream_results = stream::iter(remote_files_entries.into_iter())
+            .map(|remote_file_entry| {
+                let cache = object_storage_cache.clone();
+                let fs_accessor = filesystem_accessor.clone();
+                async move {
+                    let result = cache
+                        .get_cache_entry(
+                            remote_file_entry.file_id,
+                            &remote_file_entry.remote_filepath,
+                            fs_accessor.as_ref(),
+                        )
+                        .await;
+                    (
+                        remote_file_entry.index,
+                        remote_file_entry.remote_filepath,
+                        result,
+                    )
+                }
+            })
+            .buffer_unordered(MAX_PARALLEL_OPERATIONS);
```

Comment:
Emm, you could do `collect` so you don't need to manually iterate through all results; but not a big issue.

Reply:
https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.collect